server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -3559,6 +3559,13 @@ public boolean isRemoteTranslogEnabled() {
         return indexSettings() != null && indexSettings().isRemoteTranslogStoreEnabled();
     }
 
+    /**
+     * @return True if settings indicate this shard is backed by a remote snapshot, false otherwise.
+     */
+    public boolean isRemoteSnapshot() {
+        return indexSettings != null && indexSettings().isRemoteSnapshot();
```

Comment:
Super minor, either `indexSettings()` or `indexSettings`, but probably not both:

```suggestion
        return indexSettings() != null && indexSettings().isRemoteSnapshot();
```